### ISSUE TITLE
Create a voxelize feature that takes (n_x, n_y, n_z) as parameters instead of just density

### ISSUE THREAD

**Issue by josephko91** [pyvista/pyvista#4312]
### Describe the feature you would like to be added.

Hi! It would be great if PyVista had a feature that allowed for the voxelization of a mesh according to a specified 3d-grid encompassing the object. The current voxelize feature allows for the adjustment of a density parameter to control the voxel size, but there is no way to directly create for example a 256 by 256 by 256 voxel grid centered on bounding box of a given mesh. For example, implementing something similar to the voxelization capabilities binvox or pyntcloud would be super useful in PyVista. It seems like perhaps this is already possible in vtk, but it's not entirely clear to me as I'm not too familiar with the inner workings of vtk. 

binvox: https://www.patrickmin.com/binvox/
pyntcloud: https://pyntcloud.readthedocs.io/en/latest/structures.html highlight=add_structure#pyntcloud.PyntCloud.add_structure

### Links to VTK Documentation, Examples, or Class Definitions.

https://vtk.org/doc/nightly/html/classvtkVoxelModeller.html

**Comment by banesullivan** [comment]
I think we'd just need to modify the `voxelize()` method to use `np.linspace` instead of `arange`. Taking that code, here is an example:

```py
import numpy as np
import pyvista as pv
from pyvista import examples

mesh = examples.download_foot_bones()
surface = mesh.extract_geometry()

nx, ny, nz = 100, 75, 50

x_min, x_max, y_min, y_max, z_min, z_max = mesh.bounds
x = np.linspace(x_min, x_max, nx)
y = np.linspace(y_min, y_max, ny)
z = np.linspace(z_min, z_max, nz)
x, y, z = np.meshgrid(x, y, z)

# Create unstructured grid from the structured grid
grid = pv.StructuredGrid(x, y, z)
ugrid = pv.UnstructuredGrid(grid)

# get part of the mesh within the mesh's bounding surface.
selection = ugrid.select_enclosed_points(surface, tolerance=0.0, check_surface=True, progress_bar=True)
mask = selection.point_data['SelectedPoints'].view(np.bool_)

# extract cells from point indices
vox = ugrid.extract_points(mask, progress_bar=True)
```

-----

Thanks for pointing out `vtkVoxelModeller` -- we should look into using this directly in PyVista.

**Comment by josephko91** [comment]
Thanks @banesullivan! Example worked perfectly and thanks for the response.

**Comment by josephko91** [comment]
Adding a related feature request here. Whenever you do get around to implementing this feature in the voxelize() method, I'd also like to request a feature to save the model as a .binvox file. 

Some additional information regarding the binvox file format: https://www.patrickmin.com/binvox/binvox.html

Let me know if you'd like me to create a separate issue for this.

**Comment by user27182** [comment]
This feature is implemented as part of #6611. E.g.:
``` python
import pyvista as pv
from pyvista import examples

surface = examples.download_foot_bones()
image = surface.voxelize_binary_mask(dimensions=(100, 75, 50))
print(image.dimensions)  # (100, 75, 50) 

# Plot as voxel cells
image.points_to_cells().threshold(0.5).plot(color=True)
```

<img width="624" alt="voxelize_binary_mask" src="https://github.com/user-attachments/assets/077eda4f-c29f-4375-9d8e-339a2c2a43c0">
